src/renderer/entities/network/lib/networkService.ts

```diff
@@ -150,14 +154,26 @@ export const useNetwork = (networkSubscription?: ISubscriptionService<ChainId>):
     const knownChainId = getKnownChain(chainId);
 
     if (knownChainId) {
+      const ScProvider = createWrappedScProvider(async () => {
+        const metadata = await getMetadata(chainId);
+
+        return metadata?.metadata;
+      });
```

Comment:
Closure passed to `createWrappedScProvider` seems to be the same in all 3 places provider is created. Maybe worth to make it part of WrappedProvider itself? For example. we could pass `chainId` and `getMetadata/IMetadataService` to wrapper provider constructor